src/common/schedulers/slurm_commands.py

```diff
@@ -403,3 +422,170 @@ def _parse_nodes_info(slurm_node_info: str) -> List[SlurmNode]:
                 log.warning("Ignoring node %s because it has an invalid name", kwargs["name"])
 
     return slurm_nodes
+
+
+def _parse_job_info(job_info_string):
+    """Parse job info into job objects."""
+    # Example input
+    # [ec2-user@ip-192-168-8-106 ~]$ sudo /opt/slurm/bin/scontrol show jobs --details | awk 'BEGIN{{RS="\n\n" ;
+    # ORS="######\n";}} {{print}}' | grep -oP '^(JobId=\S+)|(JobName=\S+)|(UserId=\S+)|(Account=\S+)|(JobState=\S+)|
+    # (RunTime=\S+)|(StartTime=\S+)|(EndTime=\S+)|(Partition=\S+)|(\\bNodeList=\S+)|(\\bNodes=\S+)|(CPU_IDs=\S+)|
+    # (\\bGres=\S+)|(######)'
+    # JobId=39
```

Comment:
I'd say reduce this to 2 or 3 examples unless you think they are all necessary?

Reply:
yeah, updated the code